### PR TITLE
[FEATURE] Retry on non 401 subscription errors

### PR DESCRIPTION
**Pull request by aristidesstaffieri**
What
Adds `retryCount` optional param to `renewAndRetry`. This allows callers to add a number of retries to do in case of a non 401 error response.
Uses `retryCount` in subscriptions to retry failure cases.
Refactors and adds tests for `renewAndRetry`.

Why
In the edge case that we fail a subscription, we should retry a low number of times in order to guard against transient errors.

### CODE REVIEW

File: src/service/mercury/index.ts [hunk 1]
```diff
@@ -154,33 +154,29 @@ export class MercuryClient {
...
         await this.renewMercuryToken(network);
         this.logger.info("renewed expired jwt");
         return await method();
       }
 
-      const _error = JSON.stringify(error);
-      this.logger.error(_error);
-      throw new Error(_error);
+      // Retry in non 401 cases
+      if (retryCount) {
```
Comment (aristidesstaffieri) [inline src/service/mercury/index.ts:174]: I thought about but decided against a backoff here because this will happen in the request lifecycle and is meant to handle transient errors so it felt like slowing this down here wasn't the right move.

File: src/service/mercury/index.ts [hunk 2]
```diff
@@ -245,7 +241,7 @@ export class MercuryClient {
       };
 
       const { transferFromRes, transferToRes, mintRes } =
-        await this.renewAndRetry(subscribe, network);
+        await this.renewAndRetry(subscribe, network, 5);
```
Comment (piyalbasu) [inline src/service/mercury/index.ts:244]: NIT: maybe we should set 5 as a const like `DEFAULT_RETRY_AMOUNT` to keep these in sync and give a bit of an explanation as to what the number means at a glance

File: src/service/mercury/index.test.ts [hunk 3]
```diff
@@ -7,8 +7,13 @@ import {
   pubKey,
 } from "../../helper/test-helper";
 import { transformAccountBalances } from "./helpers/transformers";
+import { ERROR_MESSAGES } from ".";
+
+describe.only("Mercury Service", () => {
```
Comment (piyalbasu) [inline src/service/mercury/index.test.ts:12]: i think this `only` was just for testing right?

**Review by piyalbasu — APPROVED** [review]

File: src/service/mercury/index.ts (hunk 2 above)
Comment (aristidesstaffieri) [reply src/service/mercury/index.ts]: yeah good call, added in 73d2f5216802f61c2784c80acb665ed24e6f813e

File: src/service/mercury/index.test.ts (hunk 3 above)
Comment (aristidesstaffieri) [reply src/service/mercury/index.test.ts]: oh thanks, removed in 73d2f5216802f61c2784c80acb665ed24e6f813e
